pkg/sql/lease.go

```diff
@@ -574,17 +580,27 @@ func (t *tableState) acquire(
 ) (*tableVersionState, error) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
-	// Wait for any existing lease acquisition.
-	t.acquireWait()
 
 	// Acquire a lease if no lease exists or if the latest lease is
-	// about to expire.
-	if s := t.mu.active.findNewest(); s == nil || s.hasExpired(timestamp) {
-		if err := t.acquireNodeLease(ctx, m, hlc.Timestamp{}); err != nil {
+	// about to expire. Repeatedly check to ensure either a lease is
+	// acquired or we receive an error. This is required for when another
+	// routine acquires a lease and releases it immediately.
+	for s := t.mu.active.findNewest(); s == nil || s.hasExpired(timestamp); s = t.mu.active.findNewest() {
+		// Wait for any existing lease acquisition.
+		t.acquireWait()
```

Comment:
i think you can have acquireWait() return a boolean indicating if it waited. If it waited here you can just ```continue``` . That way you don't need to do the check below